src/language.cpp

```diff
@@ -369,93 +368,58 @@ void LanguageBundle::InitMetaInfo(int index)
 	for (uint32 i = 0; i < nrstrings; ++i) {
 		int size = this->values.at(i).size();
 		if (_language.GetStringName(i) == nullptr && size != 0) {
-			fprintf(stderr, "Language %s has a string at undefined index %u.\n",
+			error("Language %s has a string at undefined index %u.\n",
 					this->metadata->name, i);
-			exit(1);
 		}
 		if (size > 1 && size != this->metadata->nplurals) {
-			fprintf(stderr, "Language %s has %d plurals, but string '%s' has %d.\n",
+			error("Language %s has %d plurals, but string '%s' has %d.\n",
 					this->metadata->name, this->metadata->nplurals, _language.GetStringName(i), size);
-			exit(1);
 		}
 	}
 }
 
 /**
- * Converts a double value into a utf-8 string with the appropriate separators.
- * @param dest [out] A provided buffer to write the output into.
- * @param size The size of the provided buffer.
- * @param amt The double value to be converted.
+ * Converts a money value into a utf-8 string with the appropriate separators.
+ * @param money The value in cents to be converted.
+ * @return Formatted money string.
  */
-static void MoneyStrFmt(char *dest, size_t size, double amt)
+static std::string MoneyStrFmt(int64 money)
 {
-	const std::string curr_sym = _language.GetSgText(GUI_MONEY_CURRENCY_SYMBOL);
-	size_t curr_sym_len = curr_sym.size();
+	std::string result;
+	if (money < 0) {
+		result += "\u2212";  // U+2212 Unicode Minus sign.
```

Comment:
I'd say this needs an UTF-8 encoding setting somewhere. That is, different compilers or OSes may do different things here.

Afaik `std::string` does not promise UTF-8 in the standard.

Reply:
C++ does promise to use `char` internally, and that `char` can hold any UTF8 character. What is not promised is that string literals are UTF8-encoded by default, so I prefixed this with `u8""` which guarantees correct UTF8 encoding.

Btw we already use `\u2103` for the temperature string and it has always worked fine. (Well rendering it never worked, but that's a problem with SDL_ttf which might finally go away when we move to OpenGL and Freetype…)